crates/aranya-client-capi/ctest/test_afc.c

```diff
@@ -0,0 +1,604 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <string.h>
+#include <unistd.h>
+#include <signal.h>
+#include <sys/wait.h>
+
+#ifndef ENABLE_ARANYA_PREVIEW
+# define ENABLE_ARANYA_PREVIEW 1
+#endif
+
+#include "aranya-client.h"
+#include "utils.h"
+
+/* Globals for per-daemon client connection and sync addresses.
+    Use static defaults so tests run reproducibly without runtime snprintfs. */
+static const char g_owner_uds[128] = "/tmp/afc-run-owner/uds.sock";
+static const char g_member1_uds[128] = "/tmp/afc-run-member1/uds.sock";
+static const char g_member2_uds[128] = "/tmp/afc-run-member2/uds.sock";
+static const char g_owner_sync_addr[64] = "127.0.0.1:41001";
+static const char g_member1_sync_addr[64] = "127.0.0.1:41002";
+static const char g_member2_sync_addr[64] = "127.0.0.1:41003";
+
+/* Initialize a client */
+static AranyaError init_client(Client *c, const char *name, const char *daemon_addr) {
+    AranyaError err;
+    
+    if (name) {
+        snprintf(c->name, sizeof(c->name), "%s", name);
+    } else {
+        c->name[0] = '\0';
+    }
```

Comment:
Seems okay to always require a name to be set.
```suggestion
    if (name == NULL) {
        // return/exit
    }
    snprintf(c->name, sizeof(c->name), "%s", name);
```